apps/web-api/src/tokens/token.service.ts

```diff
@@ -90,15 +92,66 @@ export class TokenService extends BaseService<Tokens, TokenDTO> {
 
   private applyFilters(
     qb: SelectQueryBuilder<Tokens>,
-    queryArgs: IGQLQueryArgs<TokenDTO>,
+    queryArgs: QueryArgs,
   ): void {
     this.select(qb);
+
     this.applyDistinctOn(qb, queryArgs);
     this.applyLimitOffset(qb, queryArgs);
+
     this.applyWhereCondition(qb, queryArgs);
+
+    this.applyAttributesFilter(qb, queryArgs);
+
     this.applyOrderCondition(qb, queryArgs);
   }
 
+  private applyAttributesFilter(
+    qb: SelectQueryBuilder<Tokens>,
+    queryArgs: QueryArgs,
+  ): void {
+    try {
+      const attributesFilter = queryArgs?.attributes_filter;
+
+      if (!Array.isArray(attributesFilter)) {
+        return;
+      }
+
+      qb.andWhere(
+        new Brackets((qb) => {
+          attributesFilter.forEach(([key, rawValue]) => {
+            if (typeof rawValue == 'object') {
+              // Text field in format {_: "value"}
+              qb.orWhere(
+                `attributes->'${key}'->'rawValue'='${JSON.stringify(
+                  rawValue,
+                )}'::jsonb`,
+              );
+            } else {
+              // Select and multiselect field
+              qb.orWhere(
+                new Brackets((qb) => {
+                  qb.where(
+                    `attributes->'${key}'->>'rawValue'='${String(rawValue)}'`,
+                  ).orWhere(
+                    // Search value in array
+                    // eslint-disable-next-line max-len
+                    `attributes->'${key}'->>'rawValue' ~ '^\\[\\s*((\\S+\\s*,\\s*)|\\s*)*(${rawValue})((\\s*,\\s*\\S+)|\\s*)*\\]$'`,
+                  );
+                }),
+              );
+            }
+          });
+        }),
+      );
+    } catch (err) {
+      // Parse json error
+      // todo: Maybe someway report problem?
```

Comment:
Sentry?

Reply:
You're right, Sentry will do the work. And by the way, there is no more JSON.parse that I tried to catch here. Removed this try ... catch.